src/app/services/session-storage/session-storage.service.spec.ts

```diff
@@ -20,6 +20,13 @@ describe('Session storage service', () => {
     expect(sessionStorage.setItem).toHaveBeenCalledWith('exampleKey', 'exampleValue');
   });
 
+  it('should remove an item after reading it when remove is set to true', () => {
+    const service = new SessionStorageService();
```

Comment:
Should we not mock this service? You are adding a side effect to the test. In fact, would it not make more sense to test the functions `storeRedirectUrl` and `redirectToStoredUrl` and see if the mock service is called from within those?

Reply:
With this, I did just copy and paste the test above and added the clause. But I agree with your wider statement that more tests should be added.

Reply:
Oops, my bad, I thought you are testing `AuthGuard`. Maybe test `AuthGuard` as well then.

Reply:
Sorry